junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/HierarchicalTestExecutor.java

```diff
@@ -158,16 +158,21 @@ private void cleanUp() {
 		private void reportDone() {
```

Comment:
Please rework the implementation so that the new node event callbacks are invoked before `cleanUp()`.

For implementation hints, please see the comment from @marcphilipp: https://github.com/junit-team/junit5/pull/1393#issuecomment-403209962

Reply:
I [changed my mind](https://github.com/junit-team/junit5/pull/1393#issuecomment-403237102)... sorry! 😬